haystack/management/commands/update_index.py

```diff
@@ -74,21 +76,24 @@ def worker(bits):
 
 
 def do_update(backend, index, qs, start, end, total, verbosity=1):
+    start_time = now()
     # Get a clone of the QuerySet so that the cache doesn't bloat up
     # in memory. Useful when reindexing large amounts of data.
     small_cache_qs = qs.all()
     current_qs = small_cache_qs[start:end]
 
     index.pre_process_data(current_qs)
 
+    # FIXME: Get the right backend.
+    backend.update(index, current_qs)
+
+    time_delta = now() - start_time
+
     if verbosity >= 2:
         if hasattr(os, 'getppid') and os.getpid() == os.getppid():
-            print("  indexed %s - %d of %d." % (start + 1, end, total))
+            logger.info("  indexed {} - {} of {} in {}.  Took {}".format(start + 1, end, total, time_delta))
```

Comment:
@arianamiri If we're talking about the OS PID, absolutely this can happen -- and will happen in QA/PROD where concurrency is > 1 for celery. For an example, this is what's running in prod for haystack: https://gist.github.com/ruebrenda/7844d0851e38a8e9280b9e7cfbc831d6

PID 17382 is the main process and all the others are the child workers (concurrency is set to 9 for this worker).

I suspect that dev environments have concurrency set to 1 (or not set at all) so there's only one worker process running.